caddyhttp/httpserver/vhosttrie.go

```diff
@@ -103,6 +107,9 @@ func (t *vhostTrie) matchHost(host string) *vhostTrie {
 // remainingPath, and returns its node.
 func (t *vhostTrie) matchPath(remainingPath string) *vhostTrie {
 	var longestMatch *vhostTrie
+	if !CaseSensitivePath {
+		remainingPath = strings.ToLower(remainingPath)
+	}
```

Comment:
Remember to document that case sensitivity handling is guaranteed for *ASCII* and *Latin* only.

Else you'd convert it using `unicode.ToLower()`, and not doing so for performance reasons should be an intentional decision.

A *complete* handling (if ever wanted) would be *normalizing* → *convert to narrow* → *to lowercase*. ;-)